osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -86,22 +87,34 @@ public ReplayInputHandler ReplayInputHandler
 
         #region Clock control
 
-        private ManualClock clock;
-        private IFrameBasedClock parentClock;
+        private readonly ManualClock manualClock;
+
+        private readonly FramedClock framedClock;
+
+        [Cached]
+        private GameplayClock gameplayClock;
+
+        private IFrameBasedClock parentGameplayClock;
+
+        [BackgroundDependencyLoader(true)]
+        private void load(OsuConfigManager config, GameplayClock clock)
+        {
+            mouseDisabled = config.GetBindable<bool>(OsuSetting.MouseDisableButtons);
+
+            if (clock != null)
+                parentGameplayClock = clock;
+        }
 
         protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            //our clock will now be our parent's clock, but we want to replace this to allow manual control.
-            parentClock = Clock;
+            // in case a parent gameplay clock isn't available, just use the parent clock.
+            if (parentGameplayClock == null)
+                parentGameplayClock = Clock;
 
+            Clock = gameplayClock;
```

Comment:
Combine in ctor? `Clock = gameplayClock = ...`

Reply:
Can't as we may need the parent clock as a fallback (editor, as discussed).